File: Year-2023/LeetCode-April-Challenge/ValidateStackSequences.js
var validateStackSequences = function (pushed, popped) {
    let top = 0, stack = [];
    for (let x of pushed) {
        stack.push(x);
        while (stack.length && stack[stack.length - 1] === popped[top]) {
            stack.pop();
            top++;
        }
    }
    return !stack.length;
};
let pushed = [1, 2, 3, 4, 5], popped = [4, 5, 3, 2, 1];
console.log(validateStackSequences(pushed, popped));

File: Year-2023/LeetCode-April-Challenge/ValidateStackSequences.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        int top = 0; stack<int> myStack;
        for(auto ch: pushed){
            myStack.push(ch);
            while(!myStack.empty() && myStack.top() == popped[top]){
                myStack.pop();
                top++;
            }
        }
        return myStack.empty();
    }
};
int main(){
    vector<int> pushed = {1,2,3,4,5}, popped = {4,5,3,2,1};
    Solution *s = new Solution();
    cout<<std::boolalpha<<s->validateStackSequences(pushed, popped)<<endl;
    return 0;
}